src/mono/System.Private.CoreLib/src/System/RuntimeType.Mono.cs

```diff
@@ -1548,15 +1548,7 @@ private void CreateInstanceCheckThis()
                             throw new MissingMethodException(SR.Format(SR.MissingConstructor_Name, FullName));
                         }
 
-                        MethodBase? invokeMethod;
-                        object? state = null;
-
-                        try
-                        {
-                            invokeMethod = binder.BindToMethod(bindingAttr, cons, ref args, null, culture, null, out state);
-                        }
-                        catch (MissingMethodException) { invokeMethod = null; }
-
+                        MethodBase? invokeMethod = binder.BindToMethod(bindingAttr, cons, ref args, null, culture, null, out object? state);
```

Comment:
Likely this was originally done to include the MissingConstructor_Name resource string along with FullName.

Do we want to re-throw with as it was before, but with the inner exception set? We can use this ctor overload: https://learn.microsoft.com/dotnet/api/system.missingmethodexception.-ctor?view=net-8.0#system-missingmethodexception-ctor(system-string-system-exception)

Reply:
I have changed to rethrow with the inner exception set. Should we catch an exception for all types and rethrow it with the type `MissingMethodException` or any other to provide `FullName`?

Reply:
https://github.com/dotnet/runtime/pull/108876#discussion_r1805237989